projects/kit/components/input-number/input-number.component.ts

```diff
@@ -0,0 +1,255 @@
+    private readonly options = inject(TUI_INPUT_NUMBER_OPTIONS);
+    private readonly numberFormat = toSignal(inject(TUI_NUMBER_FORMAT), {
+        initialValue: TUI_DEFAULT_NUMBER_FORMAT,
+    });
+
+    private readonly min = signal(this.options.min);
+    private readonly max = signal(this.options.max);
+    private readonly prefix = signal(this.options.prefix);
+    private readonly postfix = signal(this.options.postfix);
+
+    private readonly precision = computed(() =>
+        Number.isNaN(this.numberFormat().precision) ? 2 : this.numberFormat().precision,
+    );
+
+    private readonly isIntermediateState = computed(() => {
+        const value = maskitoParseNumber(
+            this.textfieldValue(),
+            this.numberFormat().decimalSeparator,
+        );
+
+        return value < 0 ? value > this.max() : value < this.min();
+    });
+
+    protected textfieldValue = signal(this.element.value || '');
+    protected override readonly transformer = this.options.valueTransformer;
+
+    protected readonly inputMode = computed(() => {
+        if (this.isIOS && this.min() < 0) {
+            // iPhone does not have minus sign if inputMode is equal to 'numeric' / 'decimal'
+            return 'text';
+        }
+
+        return this.precision() ? 'decimal' : 'numeric';
+    });
+
+    protected readonly maxLength = computed(() => {
+        const {decimalSeparator, thousandSeparator} = this.numberFormat();
+        const decimalPart =
+            !!this.precision() && this.textfieldValue().includes(decimalSeparator);
+        const precision = decimalPart ? Math.min(this.precision() + 1, 20) : 0;
+        const takeThousand = thousandSeparator.repeat(5).length;
+
+        return DEFAULT_MAX_LENGTH + precision + takeThousand;
+    });
+
+    protected readonly mask = tuiMaskito(
+        computed(({decimalMode, ...numberFormat} = this.numberFormat()) =>
+            this.computeMask({
+                ...numberFormat,
+                precision: this.precision(),
+                min: this.min(),
+                max: this.max(),
+                prefix: this.prefix(),
+                postfix: this.postfix(),
+                decimalZeroPadding: decimalMode === 'always',
+            }),
+        ),
+    );
+
+    @Input('min')
+    public set minSetter(x: number | null) {
+        this.updateMinMaxLimits(x, this.max());
+    }
+
+    @Input('max')
+    public set maxSetter(x: number | null) {
+        this.updateMinMaxLimits(this.min(), x);
+    }
+
+    // TODO(v5): replace with signal input
+    @Input('prefix')
+    public set prefixSetter(x: string) {
+        this.prefix.set(x);
+    }
+
+    // TODO(v5): replace with signal input
+    @Input('postfix')
+    public set postfixSetter(x: string) {
+        this.postfix.set(x);
+    }
+
+    public override writeValue(value: number | null): void {
+        super.writeValue(value);
+        this.textfieldValue.set(this.formatNumber(value));
```

Comment:
I wonder if there's a way to make textfieldValue a computed based on value and isIndetermediateState? Are we going to need this type of signal in other masked controls? Maybe it should be part of `TuiControl` class?